jsettlers.logic/src/main/java/jsettlers/logic/map/loading/original/OriginalMapFileContentReader.java

```diff
@@ -408,40 +444,36 @@ public String readMapQuestText() {
 		return mapQuestText;
 	}
 
-	public String readMapQuestTip() {
-
-		if (mapQuestTip != null)
+	public String readMapQuestTip() throws MapLoadException {
+		if (mapQuestTip != null) {
 			return mapQuestTip;
+		}
 
-		MapResourceInfo filePart = findResource(OriginalMapFileDataStructs.EMapFilePartType.QUEST_TIP);
+		MapResourceInfo filePart = findResource(EMapFilePartType.QUEST_TIP);
 
-		if ((filePart == null) || (filePart.size == 0))
+		if ((filePart == null) || (filePart.size == 0)) {
 			return "";
+		}
 
 		// - Decrypt this resource if necessary
-		if (!doDecrypt(filePart))
-			return "";
+		filePart.doDecrypt();
 
 		// - read Text
 		mapQuestTip = readCStrFrom(filePart.offset, filePart.size);
 
-		// System.out.println("Tip: "+ mapQuestTip);
-
 		return mapQuestTip;
 	}
 
 	// - Read some common information from the map-file
-	public void readMapInfo() {
-		MapResourceInfo filePart = findResource(OriginalMapFileDataStructs.EMapFilePartType.MAP_INFO);
+	public void readMapInfo() throws MapLoadException {
+		MapResourceInfo filePart = findResource(EMapFilePartType.MAP_INFO);
```

Comment:
The lines from 468 to 479 repeat very very often in this code. Please extract it to a method.